### PR TITLE
 update generate_catchincr_hist.py and LADAS sample documents 

### PR DESCRIPTION
**Pull request by saraqzhang**
updates for handling forcing and increment  file exchanges in application of coupled  LADAS

### CODE REVIEW

**Comment by gmao-rreichle** [comment]
@saraqzhang : Thanks for the new PR.  A couple of things:

1. I edited the documentation a bit, see [this commit](https://github.com/GEOS-ESM/GEOSldas/pull/698/commits/4b8b7823ff58281e7f97ec39037115b421e7dafd).  Please take a look and let me know if you see anything wrong with the revised text.  
2. I cannot get `generate_catchincr_hist.py` to run, it fails before and after my edits.  I get a syntax error here: `ids = collection+f'{i:04}'` (i.e., Line 82 in the latest commit).  I was able to get this to work after changing to `ids = collection+format(i,'04d')`, but then I get a syntax error in the next line: `f.write(f"'{ids}'\n")`, at which point I ran out of time.  Does this script work for you? 
3. Once the script is working, we have to remember to replace the sample file in the "sample_config_files" directory with one that has been generated by the py script.

File: src/Applications/LDAS_App/util/config/generate_catchincr_hist.py [hunk 1]
```diff
@@ -34,32 +44,32 @@
...
+            'WESNN3_INCR'    , 'CATCHINCR_e'      ,
+            'HTSNNN1_INCR'   , 'CATCHINCR_e'      ,
+            'HTSNNN2_INCR'   , 'CATCHINCR_e'      ,
+            'HTSNNN3_INCR'   , 'CATCHINCR_e'      ,
+            'SNDZN1_INCR'    , 'CATCHINCR_e'      ,
+            'SNDZN2_INCR'    , 'CATCHINCR_e'      ,
+            'SNDZN3_INCR'    , 'CATCHINCR_e'      ,
 """
 
 nens = 32
```
Comment (gmao-rreichle) [inline src/Applications/LDAS_App/util/config/generate_catchincr_hist.py:75]: this hardcoded input `nens = 32` should be moved to the top of the script, with a comment specifying that this is the number of ens members

Comment (saraqzhang) [reply src/Applications/LDAS_App/util/config/generate_catchincr_hist.py:79]: nens = 32 is the first line  of the  actual "code" ,a reasonable place to be.  It should not be among with  the declaration of "heads" and "hist_template" , which are like constant  lists or data tables. 

I pulled the py code with your edit commits,  it works for me and generates HISTORY.rc . 
I  added  back     label= """  :: """   and.   f.write(label) , for  a cleaner  separation between the collection block and the specification block in the generated HISTORY.rc  I run the 

I run the py and verifies the resulting HISTORY.rc. These two updated files ( py and HISTORY.rc.atmens ) are committed and pushed back to the branch.

**Comment by gmao-rreichle** [comment]
@saraqzhang : Thanks for the updates. I still have a couple of questions.

- Adding "::" into the py script and HISTORY.rc.atmens makes sense.  
- Re. nens: I'm thinking of nens as something (actually, the only thing) that should be configurable by the user (because the number of ensemble members is inherently configurable).  Ideally, nens would be a command line input for the py script.  But that's more work and not really necessary, as long as it's obvious to the user what nens means, and as long as it's not buried in the middle of the script.  I'd still set nens at the top of the script and add a comment about its meaning, so it's very obvious to the user.
- I tried again and still cannot run the py script.  I'm on Discover (SLES12) and am using python/GEOSpyD/Min4.11.0_py3.9_AND_Min4.8.3_py2.7 (from ESMA_env v4.23, which is specified in the components.yaml file of the associated branch (feature/saraqzhang/ladasApp4develop).  I'm running the py script in the src dir after "source g5_modules" (from ESMA_env v4.23) and I get the following error:
```
discover12:util/config ./generate_catchincr_hist.py
  File "./generate_catchincr_hist.py", line 86
    ids = collection+f'{i:04}'
                             ^
SyntaxError: invalid syntax
```
Not sure what I'm missing.

**Comment by saraqzhang** [comment]
@gmao-rreichle   it must be  the python module version.    module load python/GEOSpyD/Ana2019.03_py3.7   ( as in the comment  at the top of the py code. )

**Comment by gmao-rreichle** [comment]
> @gmao-rreichle it must be the python module version. module load python/GEOSpyD/Ana2019.03_py3.7 ( as in the comment at the top of the py code. )

Thanks, @saraqzhang. It works for me with `python/GEOSpyD/Ana2019.03_py3.7`.  But it should work with the py version that's associated with the branch.  Not sure how to fix this.  If you don't know, maybe @weiyuan-jiang can help.

**Comment by gmao-rreichle** [comment]
> Thanks, @saraqzhang. It works for me with `python/GEOSpyD/Ana2019.03_py3.7`. But it should work with the py version that's associated with the branch. Not sure how to fix this. If you don't know, maybe @weiyuan-jiang can help.

@saraqzhang -- do you have an update on the py issue?  Should I ask Weiyuan for help?

**Comment by saraqzhang** [comment]
@gmao-rreichle   I have not looked into the py version issue yet.  Weiyuan will be able to update it quickly.

**Comment by saraqzhang** [comment]
@gmao-rreichle  I have made update to the py code , now it works with the python module version in  ldas g5_module . should I push it to this branch ?

**Comment by gmao-rreichle** [comment]
> @gmao-rreichle I have made update to the py code , now it works with the python module version in ldas g5_module . should I push it to this branch ?

yes, please!  thanks

**Comment by saraqzhang** [comment]
@gmao-rreichle  just pushed.

**Review by gmao-rreichle — APPROVED** [review]
I fixed a couple of bugs in `generate_catchincr_hist.py` and added some documentation.  
PR is trivially 0-diff.

**Pull request merged**